docs/DeveloperGuide.md

```diff
@@ -271,25 +271,33 @@ the `Allocation` association class (alternative 4).
 ### Undo/redo feature
 
 #### Implementation
+The undo/redo mechanism is added to allow users of `ResiReg` to revert to a previous state
+or return to the future state (that is the previously current state), undoing or redoing 
+changes made to the state of `ResiReg`.
 
-The undo/redo mechanism is facilitated by `StatefulResiReg`. It extends `ResiReg` with an undo/redo history, for commands that
+This mechanism is facilitated by `StatefulResiReg`, an instance of which is stored inside `ModelManager`. It extends `ResiReg` with an undo/redo history, for commands that
 modify the state of ResiReg, which comprises of: students, rooms, allocations, semesters and bin items.
+
+The class diagram below gives a pictorial representation of the class structure.
+
+![UndoRedoClassDiagram](images/UndoRedoClassDiagram.png)
+
 The history is stored internally as `redoStatesStack`, `undoStatesStack` and `currState`. Additionally, it implements the following operations:
 
-- `VersionedResiReg#save()` — Saves the current residence regulation state in its history.
-- `VersionedResiReg#undo()` — Restores the previous residence regulation state from its history.
-- `VersionedResiReg#redo()` — Restores a previously undone residence regulation state from its history.
+- `StatefulResiReg#save()` — Saves the current residence regulation state in its history.
```

Comment:
Resolved. I've replaced every instance of "residence regulation" with "ResiReg". As an additional note, I've bolded usages of "ResiReg" that refer to the application instead of the class with the same name for clear distinction between the two.